src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -82,10 +82,13 @@ class Yii2 extends Client
 
 
     private $emails = [];
+
     /**
      * @return \yii\web\Application
+     *
+     * @deprecated since 2.5, will become protected in 3.0. Use getComponent() or directly access to \Yii::$app.
```

Comment:
Don't add `getComponent()` as an option please.
Since the Yii application is a singleton the `getComponent` is a useless API that should also be deprecated.
In general for Yii applications it's more natural to use the same approach for testing as for developing.

Reply:
You introduced getComponent() 9 days ago:
https://github.com/Codeception/Codeception/pull/5173/files#diff-94abe3ee46e86aa1477f9565b6d2f9f5R189
How can it be deprecated?

Reply:
The functions in the client are for use by the `Module`; these are not intended to become part of the public API otherwise we can't change any implementations without breaking backwards compatibility.

As a user you have `grabComponent`. So deprecated is the wrong word, sorry, I should have said `@internal`.

Reply:
Ok, I've marked the new API introduced in #5173 as `@internal`.
In v3.0 all internal classes should be marked this way.

I highly rely on heavy customization of testing frameworks to accomplish reliable and fast SUTs, and semver is as need as air to me.

Reply:
> In v3.0 all internal classes should be marked this way.

I agree; I've started a discussion about this in our slack channel already.